src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileDebugTargetGenerator.cs

```diff
@@ -0,0 +1,120 @@
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
+{
+
+    /// <summary>
+    /// Provides the set of debug profiles to populate the debugger dropdown.  The Property associated
+    /// with this is the ActiveDebugProfile which contains the currently selected profile, and the DebugProfiles which
+    /// is the name of the enumerator provider
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("DebugProfileProvider")]
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    [Export(typeof(IDynamicDebugTargetsGenerator))]
+    [ExportMetadata("Name", "DebugProfileProvider")]
+    internal class DebugProfileDebugTargetGenerator : ProjectValueDataSourceBase<IReadOnlyList<IEnumValue>>, IDynamicEnumValuesProvider, IDynamicDebugTargetsGenerator
+    {
+        private IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnumValue>>> _publicBlock;
+
+        // Represents the link to the launch profiles
+        private IDisposable _launchProfileProviderLink;
+
+        // Represents the link to our source provider
+        private IDisposable _debugProviderLink;
+
+        [ImportingConstructor]
+        public DebugProfileDebugTargetGenerator(UnconfiguredProject unconfiguredProject, ILaunchSettingsProvider launchSettingProvider)
+            : base(unconfiguredProject.Services)
+        {
+            LaunchSettingProvider = launchSettingProvider;
+        }
+
+        private NamedIdentity _dataSourceKey = new NamedIdentity();
+        public override NamedIdentity DataSourceKey
+        {
+            get { return _dataSourceKey; }
+        }
+
+        /// <inheritdoc/>
+        private int _dataSourceVersion;
+        public override IComparable DataSourceVersion
+        {
+            get { return _dataSourceVersion; }
+        }
+
+        /// <inheritdoc/>
+        public override IReceivableSourceBlock<IProjectVersionedValue<IReadOnlyList<IEnumValue>>> SourceBlock
+        {
+            get
+            {
+                this.EnsureInitialized();
+                return _publicBlock;
+            }
+        }
+
+        private ILaunchSettingsProvider LaunchSettingProvider{ get; }
+
+
+        /// <summary>
+        /// This providers access to the class which creates the list of browser values..
+        /// </summary>
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<Microsoft.Build.Framework.XamlTypes.NameValuePair> options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(new DebugProfileEnumValuesGenerator(this.LaunchSettingProvider));
+        }
+
+
+        protected override void Initialize()
+        {
+            var debugProfilesBlock = new TransformBlock<ILaunchSettings, IProjectVersionedValue<IReadOnlyList<IEnumValue>>>(
+                update =>
+                {
+                    // Compute the new enum values from the profile provider
```

Comment:
No. This class derives from OnceIntializedOnceDisposed which takes care of this.